activerecord/lib/active_record/autosave_association.rb

```diff
@@ -399,7 +399,15 @@ def save_collection_association(reflection)
           if records = associated_records_to_validate_or_save(association, new_record_before_save, autosave)
             if autosave
               records_to_destroy = records.select(&:marked_for_destruction?)
-              records_to_destroy.each { |record| association.destroy(record) }
+              records_to_destroy.each do |record|
+                association.destroy(record)
+              rescue ActiveRecord::RecordNotDestroyed => e
```

Comment:
Just a knit but do we typically use `e` for the error? I'm usually trying to avoid single character names, but unsure.

Reply:
I find 9 uses of `e` within `activerecord/lib/active_record/`:

```text
connection_adapters/postgresql_adapter.rb: rescue ActiveRecord::StatementInvalid => e
connection_adapters/postgresql_adapter.rb: rescue => e
connection_adapters/abstract/connection_handler.rb: rescue LoadError => e
connection_adapters/postgresql/referential_integrity.rb: rescue ActiveRecord::ActiveRecordError => e
connection_adapters/postgresql/referential_integrity.rb: rescue ActiveRecord::InvalidForeignKey => e
connection_adapters/mysql/database_statements.rb: rescue Mysql2::Error => e
migration.rb: rescue => e
schema_dumper.rb: rescue => e
callbacks.rb: rescue RecordNotDestroyed => e
```

`error` seems to be common too (18 occurrences under the same directory). Since I'm looping on `e.record.errors`, I renamed `e` as `exception` to differentiate the rescued error from the record's errors.

Reply:
No strong opinion, I was just curious if we already have an established pattern. But if there is a mixture, then it's something we can fix down the line maybe with a rubocop rule or something. Thanks for looking!